pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataTopicTailer.java

```diff
@@ -75,12 +75,10 @@ public void run() {
                 }
             }
             try {
-                Message<byte[]> msg = reader.readNext(5, TimeUnit.SECONDS);
-                if (msg != null) {
-                    this.functionMetaDataManager.processMetaDataTopicMessage(msg);
-                }
+                Message<byte[]> msg = reader.readNext();
+                this.functionMetaDataManager.processMetaDataTopicMessage(msg);
```

Comment:
I don't think we should interrupt if  in .processMetaDataTopicMessage(msg);  We should allow that to finish.  If we don't, the functionMetadataManager might be corrupted

Reply:
I also interrupting the thread does not guarantee that we actually read to the end of the topic. reader.hasMessageAvailable() is not checked when we exit

Reply:
Good point. Also the processMetaDataTopicMessage has code execution in sync blocks. So the interrupt might happen there as well.